src/core/lib/gprpp/fork.cc

```diff
@@ -63,6 +64,12 @@ class ExecCtxState {
   }
 
   void IncExecCtxCount() {
+    // EventEngine is expected to terminate all threads before fork, and so this
+    // extra work is unnecessary
+    if (grpc_event_engine::experimental::ThreadLocal::IsEventEngineThread()) {
+      gpr_atm_no_barrier_fetch_add(&count_, 1);
```

Comment:
No, @gnossen has tests with stack traces that show EventEngine threads getting stuck here. He also patched this PR into some work he's doing to debug the fork hang issue, and found this resolved the hang.

Reply:
Ok so this is papering over the problem and we expect some rarer problem to come up (and the comment needs to be fixed still - EventEngine evidently does not terminate all threads before the fork, but we believe ignoring that is better than the alternative)

Reply:
Not quite. EventEngines are expected to terminate all of their threads pre-fork (or otherwise confirm they are fork-safe). We are dealing with an intersection of fork handlers, and non-determinism around where and when callbacks are run. This code guarantees that the _blocking behavior_ of ExecCtx on fork only applies to the older fork handling code. If an EventEngine is running this bit of code, all current known implementations are going to wait on this code to finish before allowing the fork to proceed.

I don't see anything being papered over here. This older fork code is going away, ExecCtx is (hopefully, eventually) going away. This change allows fork handlers to coexist in the meantime.

Reply:
Is it the expectation then that the thread shutdown happens *after* the exec ctx completion?

Reply:
Correct. EventEngines are not aware of ExecCtxs, parking engine threads here otherwise would require we expose something to aid in engine pre-fork handling.